src/client_state.cpp

```diff
@@ -49,6 +49,12 @@ void wsrep::client_state::close()
 {
     wsrep::unique_lock<wsrep::mutex> lock(mutex_);
     debug_log_state("close: enter");
+
+    while (mode_ == m_local && is_rollbacker_active()) {
```

Comment:
Have you considered calling `do_wait_rollback_complete_and_acquire_ownership()`?
I think it would be something like this:
```
if (mode_ == m_local && state_ == s_idle) {
   do_wait_rollback_complete_and_acquire_ownership(mutex_);
}
```

Reply:
`do_wait_rollback_complete_and_acquire_ownership()` changes state to `s_exec`, so it is not a good match here.

Reply:
ok